presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java

```diff
@@ -57,10 +56,11 @@ public synchronized void initialize(URI uri, Configuration configuration)
             throws IOException
     {
         this.localCacheFileSystem = new LocalCacheFileSystem(dataTier, uriStatus -> {
-            // URIStatus is the mechanism to pass the hiveFileContext to the source filesystem
+            // CacheContext is the mechanism to pass the hiveFileContext to the source filesystem
             // hiveFileContext is critical to use to open file.
-            checkState(uriStatus instanceof AlluxioURIStatus);
-            HiveFileContext hiveFileContext = ((AlluxioURIStatus) uriStatus).getHiveFileContext();
+            CacheContext cacheContext = uriStatus.getCacheContext();
+            checkState(cacheContext instanceof PrestoCacheContext);
```

Comment:
Hi @apc999 ,  can we add a getFileOpener method in CacheContext? then we don't need to do the cast here

Reply:
Sounds good to me